tests/prague/eip7692_eof_v1/eip7069_extcall/test_gas.py

```diff
@@ -0,0 +1,129 @@
+"""
+abstract: Tests [EIP-7069: Revamped CALL instructions](https://eips.ethereum.org/EIPS/eip-7069)
+    Tests gas comsumption
+"""  # noqa: E501
+
+import pytest
+
+from ethereum_test_tools import Account, Alloc, Environment, StateTestFiller, Transaction
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+from ethereum_test_vm import Bytecode
+
+from .. import EOF_FORK_NAME
+from . import REFERENCE_SPEC_GIT_PATH, REFERENCE_SPEC_VERSION
+from .helpers import slot_cold_gas, slot_warm_gas
+
+REFERENCE_SPEC_GIT_PATH = REFERENCE_SPEC_GIT_PATH
+REFERENCE_SPEC_VERSION = REFERENCE_SPEC_VERSION
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+def gas_test(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    setup_code: Bytecode,
+    subject_code: Bytecode,
+    tear_down_code: Bytecode,
+    cold_gas: int,
+    warm_gas: int | None = None,
+):
+    """Creates a State Test to check the gas cost of a sequence of EOF code."""
+    if cold_gas <= 0:
+        raise ValueError(f"Target gas allocations (warm_gas) must be > 0, got {cold_gas}")
+    if warm_gas is None:
+        warm_gas = cold_gas
+
+    env = Environment()
+
+    sender = pre.fund_eoa(10**18)
+
+    address_baseline = pre.deploy_contract(
+        Container(sections=[Section.Code(code=setup_code + tear_down_code)])
+    )
+    address_subject = pre.deploy_contract(
+        Container(sections=[Section.Code(code=setup_code + subject_code + tear_down_code)])
+    )
+    address_legacy_harness = pre.deploy_contract(
+        code=(
+            # warm subject and baseline without executing
+            (Op.BALANCE(address_subject) + Op.POP + Op.BALANCE(address_baseline) + Op.POP)
+            # cold gas run
+            + (
+                Op.GAS
+                + Op.CALL(address=address_subject, gas=500_000)
+                + Op.POP
+                + Op.GAS
+                + Op.SWAP1
+                + Op.SUB
+            )
+            # Baseline gas run
+            + (
+                Op.GAS
+                + Op.CALL(address=address_baseline, gas=500_000)
+                + Op.POP
+                + Op.GAS
+                + Op.SWAP1
+                + Op.SUB
+            )
+            # warm gas run
+            + (
+                Op.GAS
+                + Op.CALL(address=address_subject, gas=500_000)
+                + Op.POP
+                + Op.GAS
+                + Op.SWAP1
+                + Op.SUB
+            )
+            # Store warm gas
+            + (Op.DUP2 + Op.SWAP1 + Op.SUB + Op.PUSH2(slot_warm_gas) + Op.SSTORE)
+            # store cold gas
+            + (Op.SWAP1 + Op.SUB + Op.PUSH2(slot_cold_gas) + Op.SSTORE)
+            + Op.STOP
+        )
```

Comment:
```suggestion
        ),
        evm_code_type=EVMCodeType.LEGACY,  # Needs to be legacy to use GAS opcode
```
This forces the `address_legacy_harness` to always be legacy type, even if the test is parametrized to deploy EOF code.

Reply:
Added